substratevm/src/com.oracle.svm.reflect/src/com/oracle/svm/reflect/serialize/SerializationSupport.java

```diff
@@ -0,0 +1,202 @@
+        checksumCalculator = new ChecksumCalculator();
+    }
+
+    private static void reportError(FeatureImpl.BeforeAnalysisAccessImpl access, String msgKey, String exceptionsMsg) {
+        // Checkstyle: stop
+        String option = SubstrateOptionsParser.commandArgument(NativeImageOptions.ReportUnsupportedElementsAtRuntime, "+");
+        if (!NativeImageOptions.ReportUnsupportedElementsAtRuntime.getValue()) {
+            BigBang bb = access.getBigBang();
+            bb.getUnsupportedFeatures().addMessage(msgKey, null,
+                            exceptionsMsg + "\n" + "To allow continuing compilation with above unsupported features, set " + option);
+        } else {
+            System.out.println(exceptionsMsg);
+            System.out.println("Compilation will continue because " + option +
+                            " was set. But the program may behave unexpectedly at runtime.");
+        }
+        // Checkstyle: resume
+    }
+
+    @Platforms({Platform.HOSTED_ONLY.class})
+    public Class<?> addSerializationConstructorAccessorClass(Class<?> serializationTargetClass, Long configuredChecksum, FeatureAccess access) {
+        if (serializationTargetClass.isArray() || Enum.class.isAssignableFrom(serializationTargetClass)) {
+            return null;
+        }
+        // Don't generate SerializationConstructorAccessor class for Externalizable case
+        if (Externalizable.class.isAssignableFrom(serializationTargetClass)) {
+            try {
+                Method getExternalizableConstructor = ReflectionUtil.lookupMethod(ObjectStreamClass.class, "getExternalizableConstructor", Class.class);
+                Constructor<?> c = (Constructor<?>) getExternalizableConstructor.invoke(null, serializationTargetClass);
+                return c.getDeclaringClass();
+            } catch (Exception e) {
+                throw VMError.shouldNotReachHere(e);
+            }
+        }
+
+        String targetClassName = serializationTargetClass.getName();
+        /**
+         * Using reflection to make sure code is compatible with both JDK 8 and above Reflectively
+         * call method ReflectionFactory.newConstructorForSerialization(Class) to get the
+         * SerializationConstructorAccessor instance.
+         */
+        Constructor<?> buildTimeConstructor;
+        try {
+            Class<?> reflectionFactoryClass = access.findClassByName(Package_jdk_internal_reflect.getQualifiedName() + ".ReflectionFactory");
+            Method getReflectionFactoryMethod = ReflectionUtil.lookupMethod(reflectionFactoryClass, "getReflectionFactory");
+            Object reflFactoryInstance = getReflectionFactoryMethod.invoke(null);
+            Method newConstructorForSerializationMethod = ReflectionUtil.lookupMethod(reflectionFactoryClass, "newConstructorForSerialization", Class.class);
+            buildTimeConstructor = (Constructor<?>) newConstructorForSerializationMethod.invoke(reflFactoryInstance, serializationTargetClass);
```

Comment:
`newConstructorForSerialization` will [try to find a constructor](https://github.com/graalvm/labs-openjdk-11/blob/42e00947be342877ab34f86f7fdbbb7b0421f7f9/src/java.base/share/classes/jdk/internal/reflect/ReflectionFactory.java#L473-L485) in the given class or in any of its parent: 
So it's possible that `buildTimeConstructor` is null in case no constructor in found. Even though Java generates empty constructors automatically, the [section 2.9.1 of the JVM specs](https://docs.oracle.com/javase/specs/jvms/se11/html/jvms-2.html#jvms-2.9.1) says that:
```
A class has zero or more instance initialization methods
```
This happens only for inner classes in Java but is more common in Scala. So long story short, we need to handle the `null` case here, otherwise calling `buildTimeConstructor.getDeclaringClass();` below will generate an NPE.

Reply:
Fixed by skipping the null `buildTimeConstructor`.

Reply:
Perfect. Looks good to me!